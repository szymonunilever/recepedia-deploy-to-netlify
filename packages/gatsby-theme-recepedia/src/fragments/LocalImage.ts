import { graphql } from 'gatsby';

export const query = graphql`
  fragment LocalImage on ImageSharpFluid {
    base64
    aspectRatio
    sizes
    src
    srcSet
    srcSetWebp
    srcWebp
  }
`;
