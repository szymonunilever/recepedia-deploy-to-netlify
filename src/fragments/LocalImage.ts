import { graphql } from 'gatsby';

export default () => null;

export const query = graphql`
  fragment LocalImage on ImageSharpFluid {
    base64
    aspectRatio
    sizes
    src
    srcSet
    srcSetWebp
    srcWebp
  }
`;
