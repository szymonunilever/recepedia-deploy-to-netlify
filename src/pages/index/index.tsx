import { graphql } from 'gatsby';
import React from 'react';
import Layout from 'src/components/Layout/Layout';
import SEO from 'src/components/Seo/Seo';
import { Text, TagName } from 'src/components/lib/components/Text';

import { RecipeListing } from 'src/components/lib/components/RecipeListing';
import Hero from 'src/components/lib/components/Hero';
import { findPageComponentContent } from 'src/utils';
import { RecipeItem } from 'src/components/lib/components/RecipeListing/partials';

const HomePage = ({ data }: HomePageProps) => {
  const page = data.allPage.edges[0].node;
  const components = page.components.items;
  const recipes = data.allRecipe.nodes;

  return (
    <Layout>
      <SEO title="Recepedia Home" />

      <Text tag={TagName['h1']} text={data.allPage.edges[0].node.title} />
      <section>
        <RecipeListing
          content={findPageComponentContent(
            components,
            'RecipeListing',
            'LatestAndGreatest'
          )}
          list={recipes}
        />
      </section>

      <section>
        <RecipeListing
          content={findPageComponentContent(
            components,
            'RecipeListing',
            'TopRecipes'
          )}
          list={recipes}
        />
      </section>

      <section>
        <Hero
          content={findPageComponentContent(components, 'Hero')}
          viewType="Image"
        />
      </section>
    </Layout>
  );
};

export default HomePage;

export const pageQuery = graphql`
  {
    allPage(filter: { type: { eq: "Home" } }) {
      edges {
        node {
          ...PageFields
        }
      }
    }
    allRecipe(skip: 10) {
      nodes {
        ...RecipeFields
      }
    }
  }
`;

interface HomePageProps {
  data: {
    allPage: {
      edges: Edge<PageNode>[];
    };
    allRecipe: {
      nodes: RecipeItem[];
    };
  };
}

interface Edge<T> {
  node: T;
}

interface PageNode {
  components: {
    items: {
      name: string;
      content: any;
    }[];
  };
  title: string;
  type: string;
}
