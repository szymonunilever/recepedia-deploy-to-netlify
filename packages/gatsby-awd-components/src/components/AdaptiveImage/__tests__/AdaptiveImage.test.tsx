import { mount, ReactWrapper } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';

import { AdaptiveImage, AdaptiveImageProps } from '../index';
// eslint-disable-next-line no-undef
import LocalImage = Internal.LocalImage;
import { FluidObject } from 'gatsby-image';

describe('<AdaptiveImage />', () => {
  let wrapper: ReactWrapper;
  const fluid: FluidObject = {
    aspectRatio: 1,
    src: 'path',
    srcSet: 'path',
    sizes: 'string',
  };
  const localImage: LocalImage = {
    ext: '.svg',
    childImageSharp: {
      fluid: fluid,
      fixed: 'fixed',
    },
    fields: {
      publicURL: '/',
    },
  };
  let props: AdaptiveImageProps = {
    className: 'testClass',
    alt: 'Alt',
    localImage,
    sizes: 'size',
    view: 'view',
    link: '/',
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('matches the snapshot', () => {
    wrapper = mount(<AdaptiveImage {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render JPG image', () => {
    props = { ...props, localImage: { ...localImage, ext: '.jpg' } };
    wrapper = mount(<AdaptiveImage {...props} />);
    expect(wrapper.find('img').prop('sizes')).toBe('size');
  });

  it('should render empty element', () => {
    const props: any = {};
    wrapper = mount(<AdaptiveImage {...props} />);
    expect(wrapper.html()).toBe('');
  });
});
