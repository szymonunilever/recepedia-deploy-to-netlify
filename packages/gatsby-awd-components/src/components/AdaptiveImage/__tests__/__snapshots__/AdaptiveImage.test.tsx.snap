// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdaptiveImage /> matches the snapshot 1`] = `
<Composition
  alt="Alt"
  className="testClass"
  link="/"
  localImage={
    Object {
      "childImageSharp": Object {
        "fixed": "fixed",
        "fluid": Object {
          "aspectRatio": 1,
          "sizes": "string",
          "src": "path",
          "srcSet": "path",
        },
      },
      "ext": ".svg",
      "fields": Object {
        "publicURL": "/",
      },
    }
  }
  sizes="size"
  view="view"
>
  <ForwardRef
    className="adaptive-image__link"
    to="/"
  >
    <GatsbyLink
      className="adaptive-image__link"
      innerRef={null}
      to="/"
    >
      <ForwardRef
        className="adaptive-image__link"
        getProps={[Function]}
        innerRef={[Function]}
        onClick={[Function]}
        onMouseEnter={[Function]}
        to="/"
      >
        <Location>
          <LocationProvider
            history={
              Object {
                "_onTransitionComplete": [Function],
                "listen": [Function],
                "location": Object {
                  "assign": [Function],
                  "hash": "",
                  "host": "localhost",
                  "hostname": "localhost",
                  "href": "http://localhost/",
                  "key": "initial",
                  "origin": "http://localhost",
                  "pathname": "/",
                  "port": "",
                  "protocol": "http:",
                  "reload": [Function],
                  "replace": [Function],
                  "search": "",
                  "state": null,
                  "toString": [Function],
                },
                "navigate": [Function],
                "transitioning": false,
              }
            }
          >
            <a
              aria-current="page"
              className="adaptive-image__link"
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
              style={Object {}}
            >
              <AdaptiveImage
                alt="Alt"
                className="testClass"
                link="/"
                localImage={
                  Object {
                    "childImageSharp": Object {
                      "fixed": "fixed",
                      "fluid": Object {
                        "aspectRatio": 1,
                        "sizes": "string",
                        "src": "path",
                        "srcSet": "path",
                      },
                    },
                    "ext": ".svg",
                    "fields": Object {
                      "publicURL": "/",
                    },
                  }
                }
                sizes="size"
                view="view"
              >
                <ImageContainer
                  classNames="adaptive-image adaptiveImageSvg testClass adaptive-image-svg"
                  view="view"
                >
                  <div
                    className="adaptive-image adaptiveImageSvg testClass adaptive-image-svg"
                    data-componentname="adaptive-image"
                    data-componentview="view"
                  >
                    <img
                      alt="Alt"
                      className="adaptive-image__image"
                      src="/"
                    />
                  </div>
                </ImageContainer>
              </AdaptiveImage>
            </a>
          </LocationProvider>
        </Location>
      </ForwardRef>
    </GatsbyLink>
  </ForwardRef>
</Composition>
`;
